src/main/java/org/jenkinsci/plugins/gitclient/verifier/AcceptFirstConnectionVerifier.java

```diff
@@ -18,9 +19,28 @@ public class AcceptFirstConnectionVerifier extends HostKeyVerifierFactory {
 
     @Override
     public AbstractCliGitHostKeyVerifier forCliGit(TaskListener listener) {
-        return tempKnownHosts -> {
-            listener.getLogger().println("Verifying host key using known hosts file, will automatically accept unseen keys");
-            return "-o StrictHostKeyChecking=accept-new -o HashKnownHosts=yes";
+        return (tempKnownHosts, url) -> {
+            String hostnamePort = url.getPort() > 0 ? url.getHost() + ":" + url.getPort() : url.getHost();
+            listener.getLogger().printf("Verifying host key for %s using %s, will automatically accept unseen keys %n", hostnamePort, getKnownHostsFile().toPath());
+            String strictHostKeyChecking;
+            if (!Files.exists(getKnownHostsFile().toPath())) {
+               strictHostKeyChecking = "no";
+            } else {
+
+                List<String> hostKeyLines = Files.readAllLines(getKnownHostsFile().toPath());
+                boolean isNew = hostKeyLines.stream()
+                        .map(line -> line.split("\\s+")[0])
```

Comment:
I do not beleive this correctly handles the syntax for older ssh versions (which is why we are here)

whilst newer versions will use `|1|kbrFoDAcx....` older versions do not and are handled manually by checking for a `String.equals` or `String.contains` -  but this implementation is not correct for several cases.


The fallthought would be 
`(hostname.contains(url.getHost()) && (url.getPort() <= 0 || hostname.contains(String.valueOf(url.getPort())`

however if the `url` was `10.10.20.10:20` and the entry contained was `10.10.20.10:10` then this would say it was not new when infact it is.

There is also the possibility of multiple entries with the same key

e.g.

```
github.com,192.30.252.130 ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==
```

is a valid entry.  (the same key is used by github.com and the IP address (that is actually github.com)

I do not think this code works at all correctly in these cases either.

Reply:
Thanks James, yes, there could be several tricky cases which current implementation do not take into account currently. For now Mark prefers #882 over this PR. If once decision will be changed I'll make this changes.